File: content-solutions/dfs/finish-order.js
function solution(n, tunnels) {
  const graph = Array.from({ length: n }, () => []);
  for (const [a, b] of tunnels) {
    graph[a].push(b);
    graph[b].push(a);
  }
  for (const rooms of graph) rooms.sort((x, y) => x - y);
  const visited = new Array(n).fill(false);
  const answer = [];
  function dfs(u) {
    visited[u] = true;
    for (const v of graph[u]) {
      if (!visited[v]) dfs(v);
    }
    answer.push(u);
  }
  dfs(0);
  return answer;
}

File: content-solutions/dfs/finish-order.py
def solution(n, tunnels):
    graph = [[] for _ in range(n)]
    for a, b in tunnels:
        graph[a].append(b)
        graph[b].append(a)
    for rooms in graph:
        rooms.sort()
    visited = [False] * n
    answer = []

    def dfs(u):
        visited[u] = True
        for v in graph[u]:
            if not visited[v]:
                dfs(v)
        answer.append(u)

    dfs(0)
    return answer
